frameworks/elastic/universe/config.json

```diff
@@ -899,7 +899,7 @@
         "search_remote_connect": {
           "description": "By default, any node in the cluster can act as a cross-cluster client and connect to remote clusters. The search_remote_connect setting can be set to false (defaults to true) to prevent certain nodes from connecting to remote clusters. Cross-cluster search requests must be sent to a node that is allowed to act as a cross-cluster client. Empty means do not explicity set.",
           "type": "string",
```

Comment:
It looks like starting in Elasticsearch 6.0, *the documentation says* the that configuration file parser [will not coerce strings to booleans](https://www.elastic.co/guide/en/elasticsearch/reference/6.0/breaking_60_settings_changes.html#_coercion_of_boolean_settings).

I tried it locally, and it seems to work, contrary to what the documentation suggests:

```bash
$ wget https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-6.6.1.tar.gz
...
$ tar xvf elasticsearch-6.6.1.tar.gz
...
$ ifconfig en0 | awk '$1 == "inet" {print $2}'
172.18.8.67
$ echo 'network.host: "172.18.8.67"' > elasticsearch-6.6.1/config/elasticsearch.yml
...
$ echo 'search.remote.connect: "false"' > elasticsearch-6.6.1/config/elasticsearch.yml
...
$ elasticsearch-6.6.1/bin/elasticsearch
[2019-04-05T14:59:00,217][INFO ][o.e.e.NodeEnvironment    ] [iniP_VO] using [1] data paths, mounts [[/ (/dev/disk1s1)]], net usable_space [117.7gb], net total_space [465.
6gb], types [apfs]                                                                                                                                                       
[2019-04-05T14:59:00,219][INFO ][o.e.e.NodeEnvironment    ] [iniP_VO] heap size [989.8mb], compressed ordinary object pointers [true]
[2019-04-05T14:59:00,221][INFO ][o.e.n.Node               ] [iniP_VO] node name derived from node ID [iniP_VOzRiK3b7_Js5t4gw]; set [node.name] to override
[2019-04-05T14:59:00,221][INFO ][o.e.n.Node               ] [iniP_VO] version[6.6.1], pid[58686], build[default/tar/1fd8f69/2019-02-13T17:10:04.160291Z], OS[Mac OS X/10.1
4.3/x86_64], JVM[Oracle Corporation/Java HotSpot(TM) 64-Bit Server VM/1.8.0_152/25.152-b16]                                                                              
[2019-04-05T14:59:00,221][INFO ][o.e.n.Node               ] [iniP_VO] JVM arguments [-Xms1g, -Xmx1g, -XX:+UseConcMarkSweepGC, -XX:CMSInitiatingOccupancyFraction=75, -XX:+
UseCMSInitiatingOccupancyOnly, -Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.
encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread
=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.io.tmpdir=/var/folders/sl/pl51hwc91zl8x2x53cs5sbm80000gn/T/elasticsearch-5871600029110197844, -XX
:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=data, -XX:ErrorFile=logs/hs_err_pid%p.log, -XX:+PrintGCDetails, -XX:+PrintGCDateStamps, -XX:+PrintTenuringDistribution, -XX
:+PrintGCApplicationStoppedTime, -Xloggc:logs/gc.log, -XX:+UseGCLogFileRotation, -XX:NumberOfGCLogFiles=32, -XX:GCLogFileSize=64m, -Des.path.home=/Users/murilo/git/elasti
csearch/elasticsearch-6.6.1, -Des.path.conf=/Users/murilo/git/elasticsearch/elasticsearch-6.6.1/config, -Des.distribution.flavor=default, -Des.distribution.type=tar]    
[2019-04-05T14:59:01,732][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [aggs-matrix-stats]
[2019-04-05T14:59:01,732][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [analysis-common]
[2019-04-05T14:59:01,732][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [ingest-common]
[2019-04-05T14:59:01,732][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [lang-expression]
[2019-04-05T14:59:01,732][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [lang-mustache]
[2019-04-05T14:59:01,732][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [lang-painless]
[2019-04-05T14:59:01,732][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [mapper-extras]
[2019-04-05T14:59:01,732][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [parent-join]
[2019-04-05T14:59:01,733][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [percolator]
[2019-04-05T14:59:01,733][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [rank-eval]
[2019-04-05T14:59:01,733][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [reindex]
[2019-04-05T14:59:01,733][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [repository-url]
[2019-04-05T14:59:01,733][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [transport-netty4]
[2019-04-05T14:59:01,733][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [tribe]
[2019-04-05T14:59:01,733][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-ccr]
[2019-04-05T14:59:01,733][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-core]
[2019-04-05T14:59:01,768][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-deprecation]
[2019-04-05T14:59:01,768][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-graph]
[2019-04-05T14:59:01,768][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-ilm]
[2019-04-05T14:59:01,769][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-logstash]
[2019-04-05T14:59:01,769][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-ml]
[2019-04-05T14:59:01,769][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-monitoring]
[2019-04-05T14:59:01,769][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-rollup]
[2019-04-05T14:59:01,769][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-security]
[2019-04-05T14:59:01,769][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-sql]
[2019-04-05T14:59:01,769][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-upgrade]
[2019-04-05T14:59:01,806][INFO ][o.e.p.PluginsService     ] [iniP_VO] loaded module [x-pack-watcher]
[2019-04-05T14:59:01,806][INFO ][o.e.p.PluginsService     ] [iniP_VO] no plugins loaded
[2019-04-05T14:59:04,830][INFO ][o.e.x.s.a.s.FileRolesStore] [iniP_VO] parsed [0] roles from file [/Users/murilo/git/elasticsearch/elasticsearch-6.6.1/config/roles.yml]
[2019-04-05T14:59:05,174][INFO ][o.e.x.m.p.l.CppLogMessageHandler] [iniP_VO] [controller/58721] [Main.cc@109] controller (64 bit): Version 6.6.1 (Build a033f1b9679cab) Co
pyright (c) 2019 Elasticsearch BV                                                                                                                                        
[2019-04-05T14:59:05,496][DEBUG][o.e.a.ActionModule       ] [iniP_VO] Using REST wrapper from plugin org.elasticsearch.xpack.security.Security
[2019-04-05T14:59:05,690][INFO ][o.e.d.DiscoveryModule    ] [iniP_VO] using discovery type [zen] and host providers [settings]
[2019-04-05T14:59:06,416][INFO ][o.e.n.Node               ] [iniP_VO] initialized
[2019-04-05T14:59:06,417][INFO ][o.e.n.Node               ] [iniP_VO] starting ...
[2019-04-05T14:59:06,575][INFO ][o.e.t.TransportService   ] [iniP_VO] publish_address {172.18.8.67:9300}, bound_addresses {172.18.8.67:9300}
[2019-04-05T14:59:06,592][INFO ][o.e.b.BootstrapChecks    ] [iniP_VO] bound or publishing to a non-loopback address, enforcing bootstrap checks
[2019-04-05T14:59:09,657][INFO ][o.e.c.s.MasterService    ] [iniP_VO] zen-disco-elected-as-master ([0] nodes joined), reason: new_master {iniP_VO}{iniP_VOzRiK3b7_Js5t4gw}
{uDG354iCQ6S8IS7bQeKGzw}{172.18.8.67}{172.18.8.67:9300}{ml.machine_memory=17179869184, xpack.installed=true, ml.max_open_jobs=20, ml.enabled=true}                       
[2019-04-05T14:59:09,662][INFO ][o.e.c.s.ClusterApplierService] [iniP_VO] new_master {iniP_VO}{iniP_VOzRiK3b7_Js5t4gw}{uDG354iCQ6S8IS7bQeKGzw}{172.18.8.67}{172.18.8.67:93
00}{ml.machine_memory=17179869184, xpack.installed=true, ml.max_open_jobs=20, ml.enabled=true}, reason: apply cluster state (from master [master {iniP_VO}{iniP_VOzRiK3b7_
Js5t4gw}{uDG354iCQ6S8IS7bQeKGzw}{172.18.8.67}{172.18.8.67:9300}{ml.machine_memory=17179869184, xpack.installed=true, ml.max_open_jobs=20, ml.enabled=true} committed versi
on [1] source [zen-disco-elected-as-master ([0] nodes joined)]])                                                                                                         
[2019-04-05T14:59:09,701][INFO ][o.e.h.n.Netty4HttpServerTransport] [iniP_VO] publish_address {172.18.8.67:9200}, bound_addresses {172.18.8.67:9200}
[2019-04-05T14:59:09,702][INFO ][o.e.n.Node               ] [iniP_VO] started
[2019-04-05T14:59:09,889][WARN ][o.e.x.s.a.s.m.NativeRoleMappingStore] [iniP_VO] Failed to clear cache for realms [[]]
[2019-04-05T14:59:09,934][INFO ][o.e.l.LicenseService     ] [iniP_VO] license [f486b5b4-cb56-4e12-a408-67264964897f] mode [basic] - valid
```

Even with `network.host` configured, which starts the Elasticsearch process in "production mode".

As far as I understand, if the documentation was true the node would have failed to initialize.

I also queried the settings to check if they were using pure booleans, but it looks like they aren't:

```bash
$ curl -s "localhost:9200/_cluster/settings?include_defaults&format=yaml&flat_settings=true" | grep ': false' | wc -l
0
$ curl -s "localhost:9200/_cluster/settings?include_defaults&format=yaml&flat_settings=true" | grep ': "false"'
  action.destructive_requires_name: "false"
  action.master.force_local: "false"
  bootstrap.memory_lock: "false"
  client.transport.ignore_cluster_name: "false"
  client.transport.sniff: "false"
  cluster.blocks.read_only: "false"
  cluster.blocks.read_only_allow_delete: "false"
  cluster.remote.connect: "false"
  cluster.routing.allocation.same_shard.host: "false"
  cluster.routing.use_adaptive_replica_selection: "false"
  discovery.zen.fd.connect_on_network_disconnect: "false"
  discovery.zen.master_election.ignore_non_master_pings: "false"
  http.cors.allow-credentials: "false"
  http.cors.enabled: "false"
  http.reset_cookies: "false"
  indices.analysis.hunspell.dictionary.ignore_case: "false"
  indices.analysis.hunspell.dictionary.lazy: "false"
  indices.queries.cache.all_segments: "false"
  indices.query.query_string.analyze_wildcard: "false"
  no.model.state.persist: "false"
  node.enable_lucene_segment_infos_trace: "false"
  node.portsfile: "false"
  repositories.fs.compress: "false"
  script.painless.regex.enabled: "false"
  search.low_level_cancellation: "false"
  search.remote.connect: "false"
  transport.compress: "false"
  transport.tcp.compress: "false"
  tribe.blocks.metadata: "false"
  tribe.blocks.write: "false"
  xpack.monitoring.collection.enabled: "false"
  xpack.monitoring.collection.index.recovery.active_only: "false"
  xpack.security.audit.enabled: "false"
  xpack.security.audit.index.events.emit_request_body: "false"
  xpack.security.audit.logfile.emit_node_host_address: "false"
  xpack.security.audit.logfile.emit_node_host_name: "false"
  xpack.security.audit.logfile.events.emit_request_body: "false"
  xpack.security.audit.logfile.prefix.emit_node_host_address: "false"
  xpack.security.audit.logfile.prefix.emit_node_host_name: "false"
  xpack.security.authc.token.enabled: "false"
  xpack.security.fips_mode.enabled: "false"
  xpack.security.http.ssl.enabled: "false"
  xpack.security.transport.ssl.enabled: "false"
  xpack.watcher.encrypt_sensitive_data: "false"
  xpack.watcher.require_manual_start: "false"
```

So I guess it should work the way it is.

Reply:
Great! Let's merge this 8)